runtime/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -58,15 +58,15 @@ class ProtoBuf(context: SerialModule = EmptyModule) : AbstractSerialFormat(conte
         override fun SerialDescriptor.getTag(index: Int) = this.getProtoDesc(index)
 
         @Suppress("UNCHECKED_CAST", "NAME_SHADOWING")
-        override fun <T> encodeSerializableValue(serializer: SerializationStrategy<T>, value: T) {
+        override fun <T> encodeSerializableValue(serializer: SerializationStrategy<T>, value: T) = when {
             // encode maps as collection of map entries, not merged collection of key-values
-            if (serializer.descriptor is MapLikeDescriptor) {
+            serializer.descriptor is MapLikeDescriptor -> {
                 val serializer = (serializer as MapLikeSerializer<Any?, Any?, T, *>)
                 val mapEntrySerial = MapEntrySerializer(serializer.keySerializer, serializer.valueSerializer)
                 HashSetSerializer(mapEntrySerial).serialize(this, (value as Map<*, *>).entries)
-            } else {
-                serializer.serialize(this, value)
             }
+            serializer.descriptor == ByteArraySerializer.descriptor -> encoder.writeBytes(value as ByteArray, popTag().first)
```

Comment:
non-referential comparison is used on purpose?

Reply:
Yes. Since `ByteArraySerializer` is not an object (it has only companion object for convenience, but itself it is a regular class), the descriptor can be instantiated multiple times.